### DISCUSSION TITLE
Support for Custom Icons for Wallet Providers in Web3Modal

### DISCUSSION THREAD

**Discussion by jeyem** [WalletConnect/walletconnect-monorepo#3335] [Developer Support] [answered]
I've been working on adding new wallet connectors, specifically for Flint Wallet and two others. While I've been successful in creating and implementing the connectors, I've found that there doesn't seem to be an out-of-the-box way to assign custom icons to these wallet providers in the Web3Modal interface.

Having distinct wallet icons in the modal is crucial for a seamless and intuitive user experience. Without them, users might find the interface confusing, especially when multiple custom connectors are in use.

Could you please provide guidance on how to assign custom wallet provider icons? If this feature isn't currently supported, would it be possible to consider adding it in a future update?

Your assistance and considerations are greatly appreciated.

Best regards,

**Comment by glitch-txs** [comment, marked as answer]
You can achieve this by adding this option. The key of the image in the object is the id of the wallet's connector.

https://docs.walletconnect.com/2.0/web/web3modal/react/wagmi/custom-wallets#custom-wallet-images

**Reply by jeyem** [reply]
Thanks, Fixed
